### PR TITLE
optimize qwen1.5/2 memory usage when running long input with fp16

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

1. remove converting `logits` to fp32 to optimize qwen1.5/2 memory usage when running long input with fp16
2. current `convert_forward` cannot replace `forward` method of the outermost model, so fix it

### CODE REVIEW

**Review by rnwang04 — APPROVED** [review]
LGTM
